### ISSUE TITLE
Included lsblk version prevents aoe support

### ISSUE THREAD

**Issue by chrstphrchvz** [rockstor/rockstor-core#1219]
(Part 2 for supporting aoe as storage backend; part 1 was #867 - support block devices with up to 64 characters--previous limit was 10.)

CentOS 7 currently has `lsblk` version 2.23.2. There is an issue (resolved as of `lsblk` 2.27, cf. https://github.com/karelzak/util-linux/issues/233) that prevents rockstor from using ATA over Ethernet disks: the `aoe` kernel module puts its disks in a directory `/dev/etherd/`, but `lsblk` returns `NAME`s for devices not directly in `/dev/` apparently with an `!`  as the delimiter:

```
[root@rockstor ~]# lsblk -P -o NAME /dev/etherd/e6.6
NAME="etherd!e6.6"
```

This affects rockstor's use of `lsblk` output for commands such as `smartctl` (which I don't think will work anyways for aoe) and `mkfs.btfrs` because they will complain there is no device (e.g.) `/dev/etherd!e6.6`:

(from `/opt/rockstor/var/log/rockstor.log`)

```
ERROR [storageadmin.views.disk:186] Error running a command. cmd = ['/usr/sbin/smartctl', '--info', '/dev/etherd!e6.6']. rc = 1. stdout = ['smartctl 6.2 2013-07-26 r3841 [x86_64-linux-4.4.5-1.el7.elrepo.x86_64] (local build)', 'Copyright (C) 2002-13, Bruce Allen, Christian Franke, www.smartmontools.org', '', '/dev/etherd!e6.6: Unable to detect device type', 'Please specify device type with the -d option.', '', 'Use smartctl -h to get a usage summary', '', '']. stderr = ['']
Traceback (most recent call last):
  File "/opt/rockstor/src/rockstor/storageadmin/views/disk.py", line 184, in _update_disk_state
    do.smart_available, do.smart_enabled = smart.available(do.name)
  File "/opt/rockstor/src/rockstor/system/smart.py", line 310, in available
    [SMART, '--info', ('/dev/%s' % get_base_device(device))])
  File "/opt/rockstor/src/rockstor/system/osi.py", line 92, in run_command
    raise CommandException(cmd, out, err, rc)
CommandException: Error running a command. cmd = ['/usr/sbin/smartctl', '--info', '/dev/etherd!e6.6']. rc = 1. stdout = ['smartctl 6.2 2013-07-26 r3841 [x86_64-linux-4.4.5-1.el7.elrepo.x86_64] (local build)', 'Copyright (C) 2002-13, Bruce Allen, Christian Franke, www.smartmontools.org', '', '/dev/etherd!e6.6: Unable to detect device type', 'Please specify device type with the -d option.', '', 'Use smartctl -h to get a usage summary', '', '']. stderr = ['']

…


ERROR [storageadmin.util:47] exception: Error running a command. cmd = ['/sbin/mkfs.btrfs', '-f', '-d', 'raid1', '-m', 'raid1', '-L', 'vbox-aoe', '/dev/etherd!e6.6', '/dev/etherd!e6.7']. rc = 1. stdout = ['btrfs-progs v4.4.1', 'See http://btrfs.wiki.kernel.org for more information.', '', '']. stderr = ["Failed to check size for '/dev/etherd!e6.6': No such file or directory", '']
Traceback (most recent call last):
  File "/opt/rockstor/src/rockstor/rest_framework_custom/generic_view.py", line 40, in _handle_exception
    yield
  File "/opt/rockstor/src/rockstor/storageadmin/views/pool.py", line 273, in post
    add_pool(p, dnames)
  File "/opt/rockstor/src/rockstor/fs/btrfs.py", line 64, in add_pool
    out, err, rc = run_command(cmd)
  File "/opt/rockstor/src/rockstor/system/osi.py", line 92, in run_command
    raise CommandException(cmd, out, err, rc)
```

Is it desirable to backport a more recent `lsblk` into rockstor for this purpose, and/or does this constitute a system modification the user must perform at their risk?

A possible workaround that may involve a little creativity would be to create loopback devices for any disks not directly in `/dev/`, ensuring that the assignments are persistent.

EDIT: I came across the other issues/discussions where alternatives to `lsblk` are being considered (issue #1052, pr #786). `blkid` is another alternative, at least for addressing this issue; not sure about udev.

**Comment by chrstphrchvz** [comment]
I have not used Rockstor on CentOS 7 in some time; I am not aware whether this issue was ever fixed in util-linux provided by CentOS 7. This could be closed as won't-fix due to Rockstor migrating to openSUSE.

**Comment by phillxnet** [comment]
@chrstphrchvz Thanks for the house keeping prompt here.
Closing not as wont fix as our move to openSUSE Leap 15.2 is a form of fix so lets close this as:

Addressed by 'Built on openSUSE' endeavour re:
```
rleap15-2:~ # lsblk --version
lsblk from util-linux 2.33.1
```

Thanks for sharing your initial research on this and apologies for not following this one up. We have had very little interest in ATA over ethernet functionality and our OS migration has been all consuming on developer resources. But definitely for the good of the project as a whole it seems. Especially given more recent announcements re CentOS.
